ДЗ №2

e-shiryaev opened the pull request:
Домашнее задание 2.

Что было сделано:
- Установил material-ui и попрбовал работу с разными компонентами (пока не накладывая собственных стилей). Добавил тему оформления loft-taxi-mui-theme
- Добавил карту mapbox. Используя второй дизайн, рассудил, что карта должна быть на всем проекте. Поэтому инициализирую её 1 раз при входе в приложение и больше не трогаю (мб из-за последующих заданий будет иначе реализовано). Для инициализации освоена работа с ref и с portals
- Прописал PropTypes своим компонентам
- Разобравшись как работает контекст - я изменил "архитектуру" проекта. Теперь по состоянию isLoggedIn переключаю страницы на авторизованную часть и не авторизованную. А уже на этих страницах по состоянию переключаю отображаемые формы. Тем самым имеем не одну точку входа на 4 страницы, а первая точка входа переводит на две, вторая еще на две.
- Реализованы методы login & logout. Login реализована только на стартовой форме, при этом проверяется заполнены ли поля ввода емейла и пароля. Если не заполнены - помечаем поля ошибочными и вход не происходит. Если заполнены - заходим и запоминаем поля в состоянии контекста. Вход при форме регистрации происходит без логики, а при выходе - очищаем состояние контекста.
- Написаны два простых теста проверяющих навигацию. Переключение формы авторизации/регистрации и формы заказа такси/профиля. При попытке написать тест проверяющий работу авторизации столкнулся с двумя проблемами:
1. Не разобрался как обращаться к элементам MUI (в частности заносить поля в текстовое поле), т.к. data-testid накладывается на обертку, а не на input
2. Код из примера по проверке "существует ли объект" падает в ошибку
Сейчас этот тест в файл Auth.test.js и покрыт комментариями. Хочется получить обратную связь у наставника, после чего доделаю на следующей неделе.

Commit:
Домашнее задание 2.

Что было сделано:
- Установил material-ui и попрбовал работу с разными компонентами (пока не накладывая собственных стилей)
- Добавил карту mapbox. Используя второй дизайн, рассудил, что карта должна быть на всем проекте. Поэтому инициализирую её 1 раз при входе в приложение и больше не трогаю (мб из-за последующих заданий будет иначе реализовано). Для инициализации освоена работа с ref и с portals
- Прописал PropTypes своим компонентам
- Разобравшись как работает контекст - я изменил "архитектуру" проекта. Теперь по состоянию isLoggedIn переключаю страницы на авторизованную часть и не авторизованную. А уже на этих страницах по состоянию переключаю отображаемые формы. Тем самым имеем не одну точку входа на 4 страницы, а первая точка входа переводит на две, вторая еще на две.
- Реализованы методы login & logout. Login реализована только на стартовой форме, при этом проверяется заполнены ли поля ввода емейла и пароля. Если не заполнены - помечаем поля ошибочными и вход не происходит. Если заполнены - заходим и запоминаем поля в состоянии контекста. Вход при форме регистрации происходит без логики, а при выходе - очищаем состояние контекста.
- Написаны два простых теста проверяющих навигацию. Переключение формы авторизации/регистрации и формы заказа такси/профиля. При попытке написать тест проверяющий работу авторизации столкнулся с двумя проблемами:
1. Не разобрался как обращаться к элементам MUI (в частности заносить поля в текстовое поле), т.к. data-testid накладывается на обертку, а не на input
2. Код из примера по проверке "существует ли объект" падает в ошибку
Сейчас этот тест в файл Auth.test.js и покрыт комментариями. Хочется получить обратную связь у наставника, после чего доделаю на следующей неделе.

bread-bread commented:
карта тебе на все страницы не нужна, она нужна только после того как пользователь авторизовался, зачем лишние нагрузки на странице авторизации.
В компонентах MUI в API есть описание какие у них могут быть параметры, так что для TextField есть props inputProps через который можно достучаться до нативного инпута

e-shiryaev commented:
> карта тебе на все страницы не нужна, она нужна только после того, как пользователь авторизовался, зачем лишние нагрузки на страницу авторизации. В компонентах MUI в API есть описание, какие у них могут быть параметры, так что для TextField есть props inputProps через который можно достучаться до нативного ввода

С картой логично. Сперва подумал, что при авторизации на макете 2 уже стоит карта. Сейчас вижу, что это просто картинка имитирующая карту.

За inputProps спасибо) Зацеплюсь и доделаю

А что на счет проверки "отрисовался ли элемент"? Взял код с примера, а у меня в итоге ошибки на этой строчке

PS: мне нужно сегодня переделать замечания или я уже в следующем ДЗ их устраняю?

Commit:
ДЗ 3й недели

- Контекст авторизации переписан на Redux. Методы авторизации, логаут, получения/сохранения инфы о кредитки.
- Реализован роутинг между страницами
- Реализованы методы работы с сервером: авторизация, получение инфы о кредитке, сохранение инфы о кредитке
- Ошибка авторизации/сохранения выводится пользователю
- Вся работа с сервером через middleware (тамже идет записть в localStorage)
- Благодаря роутингу - удалил лишние файлы с уже ненужными компонентами
- Вынес файлы css в отдельную папку
- Доверстал проект так, чтобы он был похож на макет (остались мелочи с иконками и картинкой карты)
- Вынес карту только на страницу "Карта" и "Профиль", как реализовано в макете
- Не смог написать тесты =(

Вопросы:
1. Хотел подписаться на событие изменения localStorage, чтобы выбрасывать событие LOG_IN/LOG_OUT и менять storе приложения но не придумал где это лучше прописать
2. Тесты.. С прошлой недели тесты написал, они заработали. Спасибо) Но потом я сделал задания этой недели и теперь всё сломалось.

При попытке запустить render(<App/>); получаю ошибку:

Error: Could not find "store" in the context of "Connect(App)". Either wrap the root component in a <Provider>, or pass a custom React context provider to <Provider> and the corresponding React context consumer to Connect(App) in connect options.

При попытке выполнить render(<Provider store={store}><App/></Provider>); такая же ошибка
